### PR TITLE
[RFR] Use package scripts

### LINKED ISSUE #2030: run demo on Windows
env:
```
Windows 10;
node: 10.5
yarn 1.7
react-admin: 2.1.2
```
use **Powershell**

steps: 
- clone
- cd react-admin
- yarn
- cd ./example/demo
- `(set REACT_APP_DATA_PROVIDER=rest) ; (yarn start)`

result: 
```
Failed to compile
./src/fakeServer/graphql.js
Module not found: Can't resolve 'data-generator' in 'E:\code\work\react-admin\examples\demo\src\fakeServer'
```

I tried use absolute path, bu it seems so many package use this way...

How do I run this demo right? 

PS: with ubuntu 18.04, it works fine.

### PR DESCRIPTION
**Pull request by djhi**
Fixes #2030

Use package scripts compatible with all platforms (using [cross-env](https://github.com/kentcdodds/cross-env))

- [x] Build
- [x] Tests
- [x] Demos
- [x] Misc

Makefile will use those scripts.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ before_install:
     - source ~/.bashrc
     - sh -e /etc/init.d/xvfb start
     - sleep 3
+script: make test
```
Comment (djhi) [inline .travis.yml]: needed because our root package.json now has a test script and travis use it by default

File: Makefile [hunk 2]
```diff
@@ -9,110 +9,96 @@ install: package.json ## install dependencies
...
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn start
+	@yarn run-demo
 
 build-demo: ## compile the demo example to static js
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn build
+	@yarn build-demo
 
 run-graphql-demo: build ## run the demo example
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=graphql yarn start
+	@yarn run-graphql-demo
```
Comment (fzaninotto) [inline Makefile]: the task is called `run-demo-graphql` in `package.json`

File: Makefile [hunk 3]
```diff
@@ -9,110 +9,96 @@ install: package.json ## install dependencies
...
 
 run-demo: build ## run the demo example
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn start
+	@yarn run-demo
 
 build-demo: ## compile the demo example to static js
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn build
+	@yarn build-demo
 
 run-graphql-demo: build ## run the demo example
```
Comment (fzaninotto) [inline Makefile]: the `build` dependency is useless as the `package.json` command already does a `yarn build`

File: yarn.lock [hunk 4]
```diff
@@ -8758,6 +8765,29 @@ quill@~1.3.6:
     parchment "^1.1.4"
     quill-delta "^3.6.2"
 
+ra-data-graphcool@~2.0.0:
```
Comment (fzaninotto) [inline yarn.lock]: this is weird. There shouldn't be any change related to these in your PR.

File: yarn.lock [hunk 5]
```diff
@@ -8837,6 +8867,14 @@ rc@^1.0.1, rc@^1.1.6, rc@^1.2.7:
     minimist "^1.2.0"
     strip-json-comments "~2.0.1"
 
+react-admin@~2.0.0:
```
Comment (fzaninotto) [inline yarn.lock]: Same

File: Makefile [hunk 6]
```diff
@@ -9,110 +9,96 @@ install: package.json ## install dependencies
...
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=graphql yarn start
+	@yarn run-graphql-demo
 
 run-graphcool-demo: build ## run the demo example
```
Comment (fzaninotto) [inline Makefile]: same remarks

File: Makefile (hunk 2 above)
Comment (djhi) [reply Makefile]: Updated the script name accordingly

File: Makefile (hunk 3 above)
Comment (djhi) [reply Makefile]: Done

File: Makefile (hunk 6 above)
Comment (djhi) [reply Makefile]: Updated the script name accordingly and removed the build dependency

**Comment by fzaninotto** [comment]
I'm thinking: this will make the `next` merge a pain...

File: .travis.yml (hunk 1 above)
Comment (Mulli) [reply .travis.yml]: Running Windows10, powershell. Installation is bumpy.
Need to install lerna & babel (easy...)
Still fighting the error: "Module not found: Can't resolve 'data-generator' in 'C:\dev\React\react-admin\examples\demo\src\fakeServer'"

Any idea?

**Pull request merged**